### PR TITLE
[mungegithub] only wait on successful WriteComment

### LINKED ISSUE #1685: WaitForPending only on successful test trigger
stale-ci-tests calls waitforpending without checking the result of WriteComment, so if writecomment fails, it will just wait until the maxTimeOut which is 2 hours.  

```
obj.WriteComment(greenMsgBody)
ok := obj.WaitForPending(requiredContexts)
```

### PR DESCRIPTION
**Pull request by grodrigues3**
Fixes #1685

### CODE REVIEW

**Comment by foxish** [comment]
Do we also want to reduce the timeout for it to start the test? It seems to be 2 hours at the moment. 
/cc @eparis

File: mungegithub/mungers/stale-green-ci.go [hunk 1]
```diff
@@ -101,7 +101,11 @@ func (s *StaleGreenCI) Munge(obj *github.MungeObject) {
 			return
 		}
 		if time.Since(*statusTime) > staleGreenCIHours*time.Hour {
-			obj.WriteComment(greenMsgBody)
+			err := obj.WriteComment(greenMsgBody)
+			if err == nil {
```
Comment (eparis) [inline mungegithub/mungers/stale-green-ci.go]: `==` or `!=` ?

**Comment by eparis** [comment]
I think it was @ixdy who picked 2 hours. To make sure we could never queue multiple requests before the first had a reasonable chance to get started...

**Comment by eparis** [comment]
better would be a way to write the comment and move along, but just not write another comment within 2 hours. Rather than wait for the retest...

**Comment by ixdy** [comment]
https://github.com/kubernetes/contrib/pull/1022 bumped it from 1h to 2h. we were in a really bad place with Jenkins then.

**Comment by eparis** [comment]
we aren't even running mungegithub here, are we?

**Comment by grodrigues3** [comment]
@eparis no, we're not.  things are being merged manually.

**Comment by eparis** [comment]
should we?

**Pull request merged**